### PR TITLE
WIP Add build step for inertia-svelte

### LINKED ISSUE #394: svelte: Class constructor Link cannot be invoked without 'new'
### Versions:

- `@inertiajs/inertia` version: 0.8.0
- `@inertiajs/inertia-svelte` version: 0.6.0
- `@babel/core` version: 7.11.1
- `webpack` version: 4.44.2
- `@rails/webpacker` version: 5.2.1

### Describe the problem:

While trying to use the `use:inertia` action on an `<a>` tag I run into the following error:

```
Uncaught (in promise) TypeError: Class constructor Link cannot be invoked without 'new'
```

This error seems to be related to the code in the library using non-es5 features. I did a quick check with:

```
❯ npx are-you-es5 check -r .
npx: installed 10 in 2.303s
❌ @inertiajs/inertia-svelte is not ES5
❌ svelte-preprocess is not ES5
❌ tailwindcss is not ES5
```

**Note** - If I copy the current `inertia-svelte/src/link.js` into my app then things work fine, which leads me to believe that the issue is related to the use of non-es5 syntax, and node_modules not being transpiled.

That said, it's not clear to me what exactly in the inertia-svelte Link code that would be causing this, nor why this isn't a problem for other packages which certainly seem to be using non-es5 syntax.

I've tried a number of babel & webpack tweaks to try to get them to transpile the inertia-svelte code but thus far no luck. That's likely operator error on my part, but more generally I think it would be great to not have to do any of that.

### Steps to reproduce:

In my case I'm using webpack via rails webpacker gem.

1. Add `import { inertia } from '@inertiajs/inertia-svelte'`
2. Apply the inertia action to a link `<a use:inertia ...`
3. See the `Uncaught (in promise) TypeError: Class constructor Link cannot be invoked without 'new'` error

Currently I'm able to get by either using the `InertiaLink` component or by copying the action code into my repo.

### PR DESCRIPTION
**Pull request by mariojankovic**
Introducess additional changes to Chris' PR. See here: https://github.com/inertiajs/inertia/pull/439
Fixes #394.

### CODE REVIEW

File: packages/inertia-svelte/package.json [hunk 1]
```diff
@@ -19,13 +19,27 @@
...
+  },
+  "scripts": {
+    "build": "rollup -c",
+    "watch": "rollup -c -w",
+    "prepublishOnly": "npm run build"
   },
   "peerDependencies": {
-    "@inertiajs/inertia": "^0.8.0",
-    "svelte": "^3.20.0"
+    "@inertiajs/inertia": "0.8.5",
```
Comment (reinink) [inline packages/inertia-svelte/package.json:42]: Do we need `@inertiajs/inertia`  as both `devDependencies` and a `peerDependencies`? Shouldn’t it be one or the other?

Comment (mariojankovic) [reply packages/inertia-svelte/package.json:42]: Nope. Good catch.

**Comment by christoomey** [comment]
@mariojankovic this is great! Thanks for picking this up. I'm happy to consolidate efforts here and have closed my PR.

I've just tested this change locally and while it does resolve the `Class constructor Link cannot be invoked without 'new'` issue, I'm now seeing a new issue.

Using the `inertia` action is throwing an `Error: Function called outside component initialization` error, seemingly related to the [usage of createEventDispatcher](https://github.com/inertiajs/inertia/blob/da13285e33bde891ab968515d31658882a2932ce/packages/inertia-svelte/src/link.js#L9-L12) in an action ([this issue seems to cover similar things](https://github.com/sveltejs/sapper/issues/592#issuecomment-469924549)). I'm poking at different versions locally to see if I can resolve this and I'll update if I get to anything.

**Comment by reinink** [comment]
I've been able to test this locally as well using [Ping CRM Svelte](https://github.com/inertiajs/pingcrm-svelte), and it appears to be working. 🙌 

I am about to do the dev and prod builds, using either `import` (code splitting) or `require` (one big bundle).

**Comment by mariojankovic** [comment]
@christoomey that's great. I've tried including svelte within `devDependencies` and it seems the file size went up from 7kb to ~12kb (production /w terser). If that's fine you can test if that works for you.

**Comment by nathancahill** [comment]
This looks good to me!

**Comment by mariojankovic** [comment]
The PR does seem to fix this issue. I've been able to replicate the setup with no errors when using the transpiled files from the `dist` folder instead of the ones from `package.json`.

**Comment by reinink** [comment]
Closing this in favour of #549. 👍

**Pull request closed without merging**